libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -449,6 +454,82 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CWriteThenReadAsyncSuccess )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Test Function for I2C async read success
+ */
+TEST( TEST_IOT_I2C, AFQP_IotI2CReadAsyncAssisted )
+{
+    IotI2CHandle_t xI2CHandle;
+    int32_t lRetVal;
+    uint8_t ucReadBuf[16];
+    char cMsg[50]={0};
+
+    IotI2CConfig_t xI2CConfig = {
+        .ulBusFreq = testIotI2C_BAUDRATE,
+        .ulMasterTimeout =  testIotI2C_DEFAULT_TIMEOUT
+    };
+
+    if(TEST_PROTECT())
+    {
+        /* Open i2c to initialize hardware */
+        if (gIotI2cHandle[testIotI2C_INSTANCE] == NULL)
+        {
+            xI2CHandle = iot_i2c_open(testIotI2C_INSTANCE);
+            TEST_ASSERT_NOT_EQUAL(NULL, xI2CHandle);
+        }
+        else
+        {
+            xI2CHandle = gIotI2cHandle[testIotI2C_INSTANCE];
+        }
+
+        /* Set completion callback */
+        iot_i2c_set_callback(xI2CHandle, prvI2CCallback, NULL );
+
+        /* Set i2c congifuration */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetMasterConfig, &xI2CConfig);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c slave address for writing the device register */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &ucAssistedTestIotI2CSlaveAddr );
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c not stop between transaction */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSendNoStopFlag, NULL);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Write the device register address */
+        lRetVal = iot_i2c_write_sync(xI2CHandle, &xtestIotI2CDeviceRegister, sizeof(xtestIotI2CDeviceRegister));
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c slave address to read from the device register */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &ucAssistedTestIotI2CSlaveAddr );
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Read from i2c device */
+        lRetVal = iot_i2c_read_async(xI2CHandle, &ucReadBuf, sizeof(ucReadBuf));
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        lRetVal = xSemaphoreTake(xtestIotI2CSemaphore, portMAX_DELAY);
+        TEST_ASSERT_EQUAL(pdTRUE, lRetVal);
+    }
+
+    lRetVal = iot_i2c_close(xI2CHandle);
+    TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+    /* Create a string with read bytes and print it to console */
+    for(int i=0, j=0, leni=sizeof(ucReadBuf), lenj=sizeof(cMsg); i<leni && j<lenj; i++)
+    {
+        cMsg[j++]=',';
+        uint8_t upp=ucReadBuf[i]>>4, low=ucReadBuf[i]&0xF;
+        cMsg[j++]=upp+(upp>9?'A'-10:'0');
+        cMsg[j++]=low+(low>9?'A'-10:'0');
+    }
```

Comment:
Can you refactor this code to a function so that it can be reused by multiple assist test cases?

Reply:
done